File: javascript/ScratchCard.js
// javascript/ScratchCard.js
import { Product } from './Product.js';

export class ScratchCard extends Product {
    constructor(name, price, quantity) {
        super(name, price, quantity);
    }

    isExpired() {
        return false;
    }

    requiresShipping() {
        return false;
    }
}

File: Java/ScratchCard.java
public class ScratchCard extends Product {
    
    public ScratchCard(String name, double price, int quantity) {
        super(name, price, quantity);
    }
    
    @Override
    public boolean isExpired() {
        return false; // Scratch cards don't expire
    }
    
    @Override
    public boolean requiresShipping() {
        return false; // Scratch cards don't require shipping
    }
}
